paddle/pten/core/storage.h

```diff
@@ -91,12 +91,11 @@ class Storage : public intrusive_ref_counter<Storage> {
 class TensorStorage : public Storage {
  public:
   using Place = paddle::platform::Place;
-  using Allocator = deprecated::Allocator;
 
-  explicit TensorStorage(const std::shared_ptr<Allocator>& a) : alloc_(a) {}
+  explicit TensorStorage(Allocator* a) : alloc_(a) {}
 
-  TensorStorage(const std::shared_ptr<Allocator>& a, size_t size)
```

Comment:
Better use "const Allocator*" as argument?

Reply:
To allow the Tensor to modify the internal state of the Allocator, `const Allocator*` or `const shared_ptr<const Allocator>&` are not recommended here.

Reply:
I see, thx!